xarray/core/duck_array_ops.py

```diff
@@ -303,19 +303,15 @@ def f(values, axis=None, skipna=None, **kwargs):
         try:
             return func(values, axis=axis, **kwargs)
         except AttributeError:
-            if isinstance(values, dask_array_type):
-                try:  # dask/dask#3133 dask sometimes needs dtype argument
-                    # if func does not accept dtype, then raises TypeError
-                    return func(values, axis=axis, dtype=values.dtype, **kwargs)
-                except (AttributeError, TypeError):
-                    msg = "%s is not yet implemented on dask arrays" % name
-            else:
-                msg = (
-                    "%s is not available with skipna=False with the "
-                    "installed version of numpy; upgrade to numpy 1.12 "
-                    "or newer to use skipna=True or skipna=None" % name
+            if not isinstance(values, dask_array_type):
+                raise
+            try:  # dask/dask#3133 dask sometimes needs dtype argument
+                # if func does not accept dtype, then raises TypeError
+                return func(values, axis=axis, dtype=values.dtype, **kwargs)
+            except (AttributeError, TypeError):
+                raise NotImplementedError(
+                    f"{name} is not yet implemented on dask arrays"
```

Comment:
This is just removing compatibility code for old versions of NumPy, right?

Reply:
Sort of.  This code was catching *every* `AttributeError` or `TypeError`, under the assumption that the only ones that could happen were ones caused by this issue in numpy.  This made it impossible for me to find the source of an error I ran into while I was developing the main patch.

I strictly speaking could have reverted this after I got everything working, but since the check is no longer relevant, and this sort of error hiding can easily cause problems for others down the road, I figured I might as well submit the change,

Reply:
Sounds good, thanks!